Update rest webapp typescript (#295)
tigrisdata/tigris-docs: docs/guides/rest-webapp-typescript.mdx

> ## Getting Started

Reviewer: What are the Prerequisites for following this guide?

cd tigris-starter-ts
```

import StartupLocally from "@site/docs/includes/_startup_tigris_locally.mdx";

<StartupLocally />

### Create a Tigris Project

Create a Tigris project for your TypeScript RESTful web application:

```shell
tigris create project tigris_starter_ts "TypeScript RESTful App"
```

### Create a Project Application Key

After the project is created, generate an application key for it:

```shell
tigris create app_key "my app key" --project=tigris_starter_ts
```

The result will be as follows:

```
{
  "id": "dummy-id",
  "name": "my app key",
  "secret": "dummy-secret",
  "created_at": 1668493288000,
  "created_by": "google-oauth2|107496644751065904534",
  "project": "ecommerce_search"
}
```

The following table illustrates the correlation between the JSON property ID and the environment variable:

| Dialog        | Environment variable   |
| ------------- | ---------------------- |

> | URL           | `TIGRIS_URI`           |

Reviewer 2: This doesn't match the json property IDs shown above.

| Client ID     | `TIGRIS_CLIENT_ID`     |
| Client Secret | `TIGRIS_CLIENT_SECRET` |
| Name          | `TIGRIS_PROJECT`       |

> Create a `.env` file in the root directory of your project and add the credentials to it. Your `.env` file will look as follows:

Reviewer 2: We should mention something about DB_BRANCH here

```dotenv title=.env
TIGRIS_URI=api.preview.tigrisdata.cloud

> TIGRIS_PROJECT=nextjs_blog_starter_comments

Reviewer 2: Project name is incorrect

#### Insert users

> Run following commands to create two user records:

Reviewer: ```suggestion
Run the following commands to create two user records:
```

#### Read the records

> Now go ahead and confirm that the data has been persisted:

Reviewer: ```suggestion
Now, go ahead and confirm that the data has been persisted:
```

curl http://localhost:8080/products
```

### Full-text search

Tigris comes with an integrated search engine that eliminates the need to
run a separate search platform and synchronize data.

Let's search for users named "Jane":

```shell
curl http://localhost:8080/users/search \
    -X POST \
    -H 'Content-Type: application/json' \
    -d '{"query": "jane"}'
```

> Or, search for products named "avocado" and price less than 50:

Reviewer: ```suggestion
Or, search for products named "avocado" with a `price` less than `50`:
```

      res.status(412).json({ error: "Insufficient product quantity" });
      return;
    }
    const orderTotal: number = qty * product.price;

    // check balance available
    if (user.balance < orderTotal) {
      res.status(412).json({ error: "Insufficient user balance" });
      return;
    }

    // deduct balance
    await this.db.getCollection<User>(User).updateOne({
      filter: {
        userId: user.userId,
      },
      fields: {
        balance: user.balance - orderTotal,
      },
    });
    console.log("deducted user balance");

    // deduct product quantity
    await this.db.getCollection<Product>(Product).updateOne({
      filter: {
        productId: product.productId,
      },
      fields: {
        quantity: product.quantity - qty,
      },
    });
    // create order
    const order: Order = {
      orderTotal: orderTotal,
      userId: Number.parseInt(user.userId),
      productItems: [
        {
          productId: product.productId,
          quantity: qty,
          name: product.name,
          price: product.price,
        },
      ],
    };

    await this.orders.insertOne(order);
    console.log("order created");

>     res.status(200).json({ status: "Order placed successfully" });

Reviewer: More of a code comment: If a resource has been created the status code should be `201`.
